src/zenml/materializers/default_materializer.py

```diff
@@ -0,0 +1,117 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of ZenML's default materializer.
+
+This is the materializer that is used if no other materializer is registered
+for a given data type.
+"""
+
+import os
+from typing import Any, ClassVar, Tuple, Type
+
+import cloudpickle
+
+from zenml.enums import ArtifactType
+from zenml.environment import Environment
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.materializers.base_materializer import BaseMaterializer
+from zenml.utils.io_utils import (
+    read_file_contents_as_string,
+    write_file_contents_as_string,
+)
+
+logger = get_logger(__name__)
+
+DEFAULT_FILENAME = "artifact.pkl"
+DEFAULT_PYTHON_VERSION_FILENAME = "python_version.txt"
+
+
+class DefaultMaterializer(BaseMaterializer):
```

Comment:
I would rather call this `CloudpickleMaterializer` as that is what it really does. In the future this might not be our default anymore, and at that point we run into a mess keeping the old class available for backwards compatibility.

Reply:
Makes sense 👍